LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/LoRaMessage/LoRaPayload.cs

```diff
@@ -140,25 +141,21 @@ public static bool TryCreateLoRaPayload(Rxpk rxpk, out LoRaPayload loRaPayloadMe
             if (rxpk is null) throw new ArgumentNullException(nameof(rxpk));
 
             var convertedInputMessage = Convert.FromBase64String(rxpk.Data);
-            var messageType = convertedInputMessage[0];
+            var messageType = new MacHeader(convertedInputMessage[0]).MessageType;
```

Comment:
Aren't we more lenient here with respect to the conversions that we accept? Previously, if we had a `convertedInputMessage[0]` equal to let's say `51` it would not match in the `switch` statement. Now, if we have `51` it just gets shifted as `51 >> 5` and we get back a valid `MessageType`. Is that change safe?

Reply:
Edit: I now see that we have the necessary checks in the constructor. Will resolve the comment.